packages/expect-type/src/__tests__/index.test.ts

```diff
@@ -195,6 +195,60 @@ test('You can also check type guards & type assertions', () => {
   expectTypeOf(isString).guards.toBeString()
 })
 
+test('You can also check the `this` argument of functions using `.thisParam', () => {
+  type NoThisParam = (a: number) => void
+  type DateThisParam = (this: Date, a: number) => void
+  type UndefinedThisParam = (this: undefined, a: number) => void
+  type UnknownThisParam = (this: unknown, a: number) => void
+  type AnyThisParam = (this: any, a: number) => void
+
+  expectTypeOf<NoThisParam>().thisParam.toBeUnknown();
+  expectTypeOf<DateThisParam>().thisParam.toEqualTypeOf<Date>();
+  expectTypeOf<UndefinedThisParam>().thisParam.toBeUndefined();
+  expectTypeOf<UnknownThisParam>().thisParam.toBeUnknown();
+  expectTypeOf<AnyThisParam>().thisParam.toBeAny();
+})
+
+test('Of course, `.toEqualTypeOf` also distinguishes the `this` argument between functions', () => {
+  type NoThisParam = (a: number) => void
```

Comment:
I’m glad you’ve done them, but I’m not sure so many tests are needed. Should be sufficient to show that the this param matters by comparing two. After that point isn’t it just re-testing DeepBrand?

Reply:
You may be right, but still I'd prefer to have them all to be honest. I may be re-testing `DeepBrand` in a way, but I prefer the test to be fully self-contained. If this test passes, I am sure my feature works as expected. If I rely on DeepBrand implicitly, something might change in the DeepBrand implementation and the test could still pass. I want the test to be a complete specification of what the feature must do...